src/uct/ib/base/ib_md.c

```diff
@@ -27,6 +27,16 @@
                                   IBV_ACCESS_REMOTE_ATOMIC)
 #define UCT_IB_MD_RCACHE_DEFAULT_ALIGN 16
 
+typedef struct uct_ib_md_pci_info {
+    double   bw;       /* bandwidth */
+    unsigned payload;  /* payload used to data transfer */
+    unsigned overhead; /* PHY + data link layer + header + *CRC* */
+    unsigned nack;     /* number of TLC before ACK */
+    unsigned ctrl;     /* length of control TLP */
+    unsigned encoding; /* number of bits in symbol encoded, 8 - gen 1/2, 128 - gen 3 */
```

Comment:
could be float to be 8.0/10.0 or 128.0/130.0

Reply:
float? there is number of bits used in transformation, how could it ba float? :)

Reply:
i meant collapse encoding/decoding to single entry

Reply:
oh, i see
this is raw spec data (except overhead - there are tons of values), it is simpler to understand what PCI values are used having it in raw format

Reply:
ok